bin/range-finder/src/cli.rs

```diff
@@ -0,0 +1,242 @@
+                _ => Level::TRACE,
+            })
+            .finish();
+        tracing::subscriber::set_global_default(subscriber).map_err(|e| eyre!(e))?;
+        Ok(self)
+    }
+
+    /// Parse the CLI arguments and run the command
+    pub async fn run(&self) -> Result<()> {
+        // Build the pipeline
+        let cfg = Arc::new(self.rollup_config().await?);
+        let mut l1_provider = self.l1_provider()?;
+        let mut l2_provider = self.l2_provider(cfg.clone())?;
+        let attributes = self.attributes(cfg.clone(), &l2_provider, &l1_provider);
+        let blob_provider = self.blob_provider();
+        let dap = self.dap(l1_provider.clone(), blob_provider, &cfg);
+        let mut l2_cursor = self.cursor().await?;
+        let l1_tip = l1_provider
+            .block_info_by_number(l2_cursor.l1_origin.number)
+            .await
+            .expect("Failed to fetch genesis L1 block info for pipeline tip");
+        let mut pipeline = new_online_pipeline(
+            cfg,
+            l1_provider.clone(),
+            dap,
+            l2_provider.clone(),
+            attributes,
+            l1_tip,
+        );
+
+        // Run the pipeline
+        loop {
+            // If the cursor is beyond the end block, break the loop.
+            if l2_cursor.block_info.number >= self.end_block {
+                trace!(target: "loop", "Cursor is beyond the end block, breaking loop");
+                break;
+            }
+
+            // Step on the pipeline.
+            match pipeline.step(l2_cursor).await {
+                StepResult::PreparedAttributes => trace!(target: "loop", "Prepared attributes"),
+                StepResult::AdvancedOrigin => trace!(target: "loop", "Advanced origin"),
+                StepResult::OriginAdvanceErr(e) => {
+                    warn!(target: "loop", "Could not advance origin: {:?}", e)
+                }
+                StepResult::StepFailed(e) => match e {
+                    StageError::NotEnoughData => {
+                        debug!(target: "loop", "Not enough data to step derivation pipeline");
+                    }
+                    _ => {
+                        error!(target: "loop", "Error stepping derivation pipeline: {:?}", e);
+                    }
+                },
+            }
+
+            // Get the attributes if there are some available.
+            let attributes = if let Some(attributes) = pipeline.next() {
+                attributes
+            } else {
+                continue;
+            };
+
+            // Print the L1 range for this L2 Block.
+            let derived = attributes.parent.block_info.number as i64 + 1;
+            let l2_block_info = l2_provider
+                .l2_block_info_by_number(derived as u64)
+                .await
+                .expect("Failed to fetch L2 block info for pipeline cursor");
+            let origin = pipeline.origin().expect("Failed to get pipeline l1 origin");
+            println!(
```

Comment:
Can we change this to an `info!` trace?

Reply:
Having it be a print allows the user to not have to pass in the `-v` flag at all. And I think it's acceptable to use a print here since finding this output is the entire point of the binary.

Reply:
Yeah makes sense to me!